staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -176,22 +180,6 @@ func (s *Validator) validateExpressions(fldPath *field.Path, sts *schema.Structu
 				remainingBudget -= int64(*rtCost)
 			}
 		}
-		if err != nil {
-			// see types.Err for list of well defined error types
-			if strings.HasPrefix(err.Error(), "no such overload") {
-				// Most overload errors are caught by the compiler, which provides details on where exactly in the rule
-				// error was found. Here, an overload error has occurred at runtime no details are provided, so we
-				// append a more descriptive error message. This error can only occur when static type checking has
-				// been bypassed. int-or-string is typed as dynamic and so bypasses compiler type checking.
-				errs = append(errs, field.Invalid(fldPath, obj, fmt.Sprintf("'%v': call arguments did not match a supported operator, function or macro signature for rule: %v", err, ruleErrorString(rule))))
-			} else if strings.HasPrefix(err.Error(), "operation cancelled: actual cost limit exceeded") {
-				errs = append(errs, field.Invalid(fldPath, obj, fmt.Sprintf("'%v': call cost exceeds limit for rule: %v", err, ruleErrorString(rule))))
```

Comment:
were we expecting to short circuit remaining validation rules if one exceeds its per-cost limit? if so, I do think that was happening, right? We'd decrement remainingCost by the evaluation cost, and include an error here, but not truncate remainingBudget to -1, right?

Reply:
Yes. If the cost limit exceed for one expression but the overall cost limit does not exceed, we will keep validating the following rules and return error for this specific failed rule to indicate this rule exceed per expression limit. The CR validation as a whole sure will fail but all the following rules get the chance to be validated and we will return all possible errors together back to user.

Reply:
hmm, I had been expecting to short circuit further evaluations if we exceeded cost limits in any dimension

Reply:
I guess the initial idea of validation is to get all possible check failures so user could fix them at once. We could definitely short circuit further evaluations for cost exceed per expression as well. Not sure which way is better from users' perspective.

Reply:
if it were free, definitely more complete errors are better. if we're bumping up against the protections we put in place for cost limits, I think we err on the side of safety and stop evaluating

Reply:
Make sense. I have updated to short circuit further evaluation when any cost limit exceed. Would you mind to take a look? Thank you!